clients/src/main/java/org/apache/kafka/clients/consumer/KafkaConsumer.java

```diff
@@ -1581,7 +1581,27 @@ public void commitAsync(final Map<TopicPartition, OffsetAndMetadata> offsets, Of
      * Overrides the fetch offsets that the consumer will use on the next {@link #poll(Duration) poll(timeout)}. If this API
      * is invoked for the same partition more than once, the latest offset will be used on the next poll(). Note that
      * you may lose data if this API is arbitrarily used in the middle of consumption, to reset the fetch offsets
+     * <p>
+     * The next Consumer Record which will be retrieved when poll() is invoked will have the offset specified, given that
+     * a record with that offset exists (ie: it is a valid offset).
+     * <p>
+     * {@link KafkaConsumer#seekToBeginning(Collection)} will go to the first offset in the topic.
+     * seek(0) is equivalent to seekToBeginning for a topic with beginning offset 0,
+     * assuming that there is a record at offset 0 still available.
+     * {@link KafkaConsumer#seekToEnd(Collection)} is equivalent to seeking to the highest known offset + 1.
+     * <p>
+     * Seeking to the offset smaller than the log start offset or larger than the log end offset
+     * or high watermark means an invalid offset is reached.
+     * Invalid offset behaviour is controlled by
+     * the {@link ConsumerConfig AUTO_RESET_CONFIG} property.
+     * If this is set to "earliest", the next poll will return records from the starting offset.
+     * If it is set to "latest", it will seek to the last known record (similar to seekToEnd()).
+     * If it is set to "none", an {@link OffsetOutOfRangeException} will be thrown.
```

Comment:
I don't think it will link to `OffsetOutOfRangeException` with this format. Maybe we just use `code`:
* If it is set to "none", an {@code OffsetOutOfRangeException} will be thrown.